web/satellite/src/components/project/paymentMethods/NewProjectPaymentMethodComponent.vue

```diff
@@ -0,0 +1,195 @@
+
+                        <!-- Used to display form errors. -->
+                        <div id="card-errors" role="alert"></div>
+                    </div>
+                    <div class="checkbox-container" v-if="projectPaymentMethodsCount > 0">
+                        <Checkbox @setData="toggleMakeDefault"/>
+                        <h2>Make Default</h2>
+                    </div>
+                    <Button
+                            label="Save"
+                            width="135px"
+                            height="48px"
+                            :on-press="onSaveClick"/>
+                </form>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import { Component, Vue } from 'vue-property-decorator';
+    import Button from '@/components/common/Button.vue';
+    import {
+        NOTIFICATION_ACTIONS,
+        PROJECT_PAYMENT_METHODS_ACTIONS,
+        USER_PAYMENT_METHODS_ACTIONS
+    } from '@/utils/constants/actionNames';
+    import Checkbox from '@/components/common/Checkbox.vue';
+    import Stripe from '@/utils/stripe';
+
+    @Component(
+        {
+            components: {
+                Button,
+                Checkbox,
+            }
+        }
+    )
+
+    export default class NewProjectPaymentMethodComponent extends Vue {
+        private makeDefault: boolean = false;
+        private isSaveButtonEnabled: boolean = true;
+
+        public mounted(): void {
+            const stripe: Stripe = new Stripe();
+            try {
+                stripe.newCardInput(this.onStripeResponse);
+            } catch (e) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, e.toString());
+            }
+        }
+
+        private async onStripeResponse(result: any) {
+            const input = {
+                token: result.token.id,
+                makeDefault: this.makeDefault} as AddPaymentMethodInput;
+
+            const response = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.ADD, input);
+            this.isSaveButtonEnabled = true;
+            if (!response.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, response.errorMessage);
+
+                return;
+            }
+            this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Card successfully added');
+
+            const projectPaymentsResponse = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.FETCH);
+            if (!projectPaymentsResponse.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch payment methods: ' + projectPaymentsResponse.errorMessage);
+            }
+
+            const userPaymentMethodResponse = await this.$store.dispatch(USER_PAYMENT_METHODS_ACTIONS.FETCH);
+            if (!userPaymentMethodResponse.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch user payment methods: ' + userPaymentMethodResponse.errorMessage);
```

Comment:
lets use template string to prevent smth like this
Unable to fetch user payment methods: ' + +userPaymentMethodResponse.errorMessage

`Unable to fetch user payment methods: ${userPaymentMethodResponse.errorMessage}`